tests/kernel/run_ltp.pm

```diff
@@ -359,8 +352,8 @@ sub run_post_fail {
 
     $self->fail_if_running();
 
-    if ($self->{force_softfail}) {
-        $self->{result} = 'softfail';
+    if ($self->{ltp_tinfo} and get_var('LTP_KNOWN_ISSUES') and $self->{result} eq 'fail') {
+        override_known_failures($self, $self->{ltp_env}, $self->{ltp_tinfo}->runfile, $self->{ltp_tinfo}->test->{name});
```

Comment:
Are those additional arguments still needed, if they are now available through the first one (`$self`)?

Reply:
Technically not, but since it's an external function, I think it's cleaner to pass them this way than to make assumptions about `run_ltp` instance internals in `override_known_failures()`.

Reply:
Well, you added that assumptions yourself on line 303 with defining `ltp_env` and `ltp_info` and by not using the return value of the `override_known_failures`. Now you are making the some result overriding magic inside the method via `$self->record_soft_failure_result` so you still need to know about the `run_ltp` instance internals, right?

Reply:
The only assumption I'm making with `$self->record_soft_failure_result()` is that `$self` is an instance of `os-autoinst::basetest`. Nothing more. `testapi::record_soft_failure()` doesn't work when called (indirectly) from `run_ltp::run_post_fail()` because it tries to call `$self->record_soft_failure_result()` through a global variable in `os-autoinst` that was already unset after `run_ltp::run()` finished.

I don't understand what assumptions you mean regarding `ltp_env`, `ltp_info` and `force_softfail`. I'm treating those attributes as private to the `run_ltp` class. `run_post_fail()` is a member function of `run_ltp`. `override_known_failures()` is not.

Reply:
no strong opinion from anybody, merging